go/vt/sqlparser/sql.y

```diff
@@ -2061,6 +2061,10 @@ table_name:
   {
     $$ = TableName{Qualifier: $1, Name: $3}
   }
+| table_id '.' '*'
```

Comment:
This may be a little too permissive because it will allow invalid constructs like `select a from t.*`. We should probably create a `delete_table_list` and special-case that to allow `t.*`.

Reply:
Fixed that. And added and end to end test.